woocommerce/woocommerce-android
# Disable non editable fields

‼️ Don't merge until https://github.com/woocommerce/woocommerce-android/pull/6700 changes are merged

Closes: #6651

### Description


This PR disables the necessary sections when the order is not editable based on the value of the `isEditable` field and adds the `status` field to the fields who has the source of truth on the API. The `isEditable` field depends on the `status` of the order and is managed by the API, for this reason, I choose to make the `status` always synced with the API as we do with products, shipping, and fees. When the order is not editable all sections affecting the order's total are disabled, no changes such as adding a new product or a fee are allowed in a non-editable order. Notes and Customer sections are always editable without taking into account the `isEditable` field.

### Testing instructions

1. Open the Orders section -> Select an Order -> Menu (3 dots) -> Select Edit
2. Change the Order status to an editable status such as `Pending payment`
3. Check that editable sections of products, fees, and shipping are enabled
4. Change the Order status to a non-editable status such as `Processing`
5. Check that editable sections of products, fees, and shipping are disabled



### Images/gif


https://user-images.githubusercontent.com/18119390/172426239-076c28b3-b857-409f-ae76-a3798d9d5f02.mp4
